adaptive_components/lib/src/adaptive_container.dart

```diff
@@ -0,0 +1,241 @@
+  /// 12 columns per flex range.
+  ///
+  /// Learn more by visiting the Material website:
+  /// https://material.io/design/layout/responsive-layout-grid.html#breakpoints
+  final int columnSpan;
+
+  /// Align the [child] within the container.
+  ///
+  /// If non-null, the container will expand to fill its parent and position its
+  /// child within itself according to the given value. If the incoming
+  /// constraints are unbounded, then the child will be shrink-wrapped instead.
+  ///
+  /// Ignored if [child] is null.
+  ///
+  /// See also:
+  ///
+  ///  * [Alignment], a class with convenient constants typically used to
+  ///    specify an [AlignmentGeometry].
+  ///  * [AlignmentDirectional], like [Alignment] for specifying alignments
+  ///    relative to text direction.
+  final AlignmentGeometry alignment;
+
+  /// Empty space to inscribe inside the [decoration]. The [child], if any, is
+  /// placed inside this padding.
+  ///
+  /// This padding is in addition to any padding inherent in the [decoration];
+  /// see [Decoration.padding].
+  final EdgeInsetsGeometry padding;
+
+  /// The color to paint behind the [child].
+  ///
+  /// This property should be preferred when the background is a simple color.
+  /// For other cases, such as gradients or images, use the [decoration]
+  /// property.
+  ///
+  /// If the [decoration] is used, this property must be null. A background
+  /// color may still be painted by the [decoration] even if this property is
+  /// null.
+  final Color color;
+
+  /// The decoration to paint behind the [child].
+  ///
+  /// Use the [color] property to specify a simple solid color.
+  ///
+  /// The [child] is not clipped to the decoration. To clip a child to the shape
+  /// of a particular [ShapeDecoration], consider using a [ClipPath] widget.
+  final Decoration decoration;
+
+  /// The decoration to paint in front of the [child].
+  final Decoration foregroundDecoration;
+
+  /// Empty space to surround the [decoration] and [child].
+  final EdgeInsetsGeometry margin;
+
+  /// The transformation matrix to apply before painting the container.
+  final Matrix4 transform;
+
+  /// The clip behavior when [Container.decoration] has a clipPath.
+  ///
+  /// Defaults to [Clip.none].
+  final Clip clipBehavior;
+
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(
+      builder: (BuildContext context, BoxConstraints boxConstraints) {
+        if (constraints.withinAdaptiveConstraint(context)) {
+          return Container(
+            alignment: alignment,
+            padding: padding,
+            color: color,
+            decoration: decoration,
+            foregroundDecoration: foregroundDecoration,
+            transform: transform,
+            clipBehavior: clipBehavior,
+            height: height,
+            margin: margin,
+            child: child,
+          );
+        } else {
+          return LimitedBox(
```

Comment:
Hmm would just an empty `Container()` be cleaner than this? Also add a small comment explaining why this is needed.

Reply:
Done. I am using the limitedBox since it is a more efficient widget to use.